### ISSUE TITLE
Make `select`, `exclude`, `selector`, and `models` pseudo-parametrizable via skipping and `trigger_rule="none_failed"`

### ISSUE THREAD

**Issue by dwreeves** [astronomer/astronomer-cosmos#857]
Right now it is impossible to pass a `Param()` into a `DbtDag()` or `DbtTaskGroup()` that selects and de-selects models.

The reason why is pretty obvious when you think about it: these params are used to render the tasks in the DbtDag, and rendering does not have access to Jinja templating (only tasks while they are being executed have access).

That said, it could be possible to make accessible some sort of mechanism for `raise AirflowSkipException` when a task is excluded via a different mechanism than the DAG rendering. In combination with `trigger_rule="none_failed"`, this could provide a mechanism for running subsets of tasks in a DAG via Params.

Because of Airflow's internals, the selection of tasks in a DAG cannot be determined via Params, except via dynamic task mapping, which does not support task dependencies. However because skipping and `trigger_rule="none_failed"` achieves a similar behavior, this is why I refer to this as making the DbtDag "pseudo-parametrizable".

There are some real challenges here, this isn't as simple as just passing some args into the operators, but it's super-duper feasible and possible to imagine this existing as a feature.

---

There is some slightly related discussion in #520.

**Comment by pankajastro** [comment]
Hey @dwreeves, now we have support for `operator_args` to pass the airflow config per model: 
- https://astronomer.github.io/astronomer-cosmos/getting_started/custom-airflow-properties.html
- https://astronomer.github.io/astronomer-cosmos/configuration/operator-args.html

If trigger_rule were a [templated field](https://github.com/astronomer/astronomer-cosmos/blob/9405903807dabc0c6bc954296ff4a3724a959c4f/cosmos/operators/base.py#L79), would it solve the issue?

**Comment by hhhonzik** [comment]
Would this be solved by https://github.com/astronomer/astronomer-cosmos/issues/1228?

**Comment by pankajastro** [comment]
We have created an issue: https://github.com/astronomer/astronomer-cosmos/issues/1289 to solve this problem in a more generalised way, allowing us to support more templated fields and avoid the error during parsing.

**Comment by pankajastro** [comment]
@hhhonzik yes, I believe https://github.com/astronomer/astronomer-cosmos/issues/1228, https://github.com/astronomer/astronomer-cosmos/issues/1289 and this are aim the solve the same problem
